From 6a0b5cc71afca04ac0f74347d2528cdbb05cd110 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Tue, 31 Dec 2013 12:16:58 +0200
Subject: [PATCH 425/987] fmd: adapt to absence of rt code

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Use non-rt api in FMD when rt api is not available.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: I25e16f18a6f2f3bb16a80d1431b54b45db915f29
Reviewed-on: http://git.am.freescale.net:8181/7583
Reviewed-by: Mandy Lavi <Mandy.Lavi@freescale.com>
---
 drivers/net/ethernet/freescale/fman/src/xx/xx_linux.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/ethernet/freescale/fman/src/xx/xx_linux.c b/drivers/net/ethernet/freescale/fman/src/xx/xx_linux.c
index 132459b..546c68b 100644
--- a/drivers/net/ethernet/freescale/fman/src/xx/xx_linux.c
+++ b/drivers/net/ethernet/freescale/fman/src/xx/xx_linux.c
@@ -354,14 +354,22 @@ uint32_t XX_DisableAllIntr(void)
 {
     unsigned long flags;
 
+#ifdef local_irq_save_nort
     local_irq_save_nort(flags);
+#else
+    local_irq_save(flags);
+#endif
 
     return (uint32_t)flags;
 }
 
 void XX_RestoreAllIntr(uint32_t flags)
 {
+#ifdef local_irq_restore_nort
     local_irq_restore_nort((unsigned long)flags);
+#else
+    local_irq_restore((unsigned long)flags);
+#endif
 }
 
 t_Error XX_Call( uint32_t qid, t_Error (* f)(t_Handle), t_Handle id, t_Handle appId, uint16_t flags )
-- 
1.9.1

